Add ScalingGizmo

vantemoon opened the pull request:
**Description**

Implemented a scaling gizmo with the following behaviours:

- Dragging the axis handles scales the object in the direction of the corresponding axis.
- Dragging the centred cube scales the object uniformly in all directions.

Please refer to the code comments for a detailed explanation.

**Test plan**
To test the scaling gizmo, use the `src/tests/gfx-gizmos-scaling.edn` file.

sinkingsugar commented on src/gfx/gizmos/scaling_gizmo.hpp:
```diff
@@ -0,0 +1,228 @@
+    dragStartTransform = transform;
+
+    size_t index = getHandleIndex(handle);
+    SPDLOG_DEBUG("Handle {} ({}) grabbed", index, getAxisDirection(index, dragStartTransform));
```
since you finished debugging I assume , can be trace or even removed? (likewise for other gizmos)

sinkingsugar reviewed (approved):
Looks good

guusw commented:
There's some gaps between the capped handles and the center cube hitbox 
<img width="299" alt="image" src="https://github.com/fragcolor-xyz/shards/assets/16842774/69c56ddd-82b1-4124-9f53-a99b89a8a172">
Could you make them connect (at least the ones on the side), the one facing the camera can have a gap & be smaller (use the existing `angleFactor`)

vantemoon commented:
@guusw 

Thank you for your suggestions! I have addressed the following requested changes in the latest commit b5f857ea. 

- Modified the variable names to follow the camelCase convention.
- Moved the code related to the centred cube outside of the loops. The loops now solely focus on the axis handles.
- Utilised float3's index operators to reduce code duplication.

vantemoon commented on src/gfx/gizmos/scaling_gizmo.hpp:
```diff
@@ -0,0 +1,228 @@
+        handles[i].grabOffset = centeredCubeSensitivity * delta.x;
+      }
+      // Dragging the centered cube to the right increases the object's scale, while dragging it to the left decreases the object's scale.
+      float diameter = centeredCubeSensitivity * std::abs(delta.x);
```
I've tried some different ways of calculating `scaling` using `std::max()`,  `std::abs()` or combinations of the two but they all lead to some undesirable behaviours when moving the mouse across the y-axis of the screen. I'm still trying to figure out if there's a better implementation of this.

guusw commented on src/tests/gfx-gizmos-scaling.edn:
```diff
@@ -0,0 +1,82 @@
+    (GFX.Render :Steps .render-steps :View .view))))
+
+(schedule root test-wire)
+(if (run root timestep) nil (throw "Root tick failed"))
```
Don't forget to add a fixed duration back - e.g. `(run root timestep 100)` - so that CI doesn't get stuck.

vantemoon replied:
Noted. Thanks!

Yttruire commented:
Note: This branch uses the original implementations of Handle in TranslationGizmo and the ScalingGizmo. The location of the raycasting code and how handles are used have been refactored in the RotationGizmo branch. Should merge this branch into the RotationGizmo one, resolve the merge conflicts there then merge into devel.
